### ISSUE TITLE
Test failure Interop/PInvoke/Generics/GenericsTest/GenericsTest.sh

### ISSUE THREAD

**Issue by VincentBu** [dotnet/runtime#61237]
Run: [runtime 20211104.48](https://dev.azure.com/dnceng/public/_build/results?buildId=1454720&view=ms.vss-test-web.build-test-results-tab&runId=41847820&paneView=debug&resultId=102052)

Failed test:
```

CoreCLR Linux x64 Checked no_tiered_compilation @ Ubuntu.1804.Amd64.Open

- Interop/PInvoke/Generics/GenericsTest/GenericsTest.sh

CoreCLR Linux x64 Checked @ Ubuntu.1804.Amd64.Open

- Interop/PInvoke/Generics/GenericsTest/GenericsTest.sh

```
**Error message:**
```
/datadisks/disk1/work/BD200A9D/w/B95109F8/e/Interop/PInvoke/Generics/GenericsTest/GenericsTest.sh: line 379: 16185 Illegal instruction     (core dumped) $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"

Return code:      1
Raw output file:      /datadisks/disk1/work/BD200A9D/w/B95109F8/uploads/Reports/Interop.PInvoke/Generics/GenericsTest/GenericsTest.output.txt
Raw output:
BEGIN EXECUTION
/datadisks/disk1/work/BD200A9D/p/corerun -p System.Reflection.Metadata.MetadataUpdater.IsSupported=false GenericsTest.dll ''
Testing IComInterface<bool>
Testing IComInterface<char>
Testing IComInterface<double>
Testing IComInterface<float>
Testing IComInterface<long>
Testing IComInterface<uint>

Testing Nullable<bool>
Testing Nullable<char>
Testing Nullable<double>
Testing Nullable<float>
Testing Nullable<long>
Testing Nullable<uint>

Testing Point1<bool>
Testing Point1<char>
Testing Point1<double>
Testing Point1<float>
Testing Point1<long>
Testing Point1<uint>

Testing Point2<bool>
Testing Point2<char>
Testing Point2<double>
Testing Point2<float>
Testing Point2<long>
Testing Point2<uint>

Testing Point3<bool>
Testing Point3<char>
Testing Point3<double>
Testing Point3<float>
Testing Point3<long>
Testing Point3<uint>

Testing Point4<bool>
Testing Point4<char>
Testing Point4<double>
Testing Point4<float>
Testing Point4<long>
Testing Point4<uint>

Testing ReadOnlySpan<bool>
Testing ReadOnlySpan<char>
Testing ReadOnlySpan<double>
Testing ReadOnlySpan<float>
Testing ReadOnlySpan<long>
Testing ReadOnlySpan<uint>

Testing SequentialClass<bool>
Testing SequentialClass<char>
Testing SequentialClass<double>
Testing SequentialClass<float>
Testing SequentialClass<long>
Testing SequentialClass<uint>

Testing Span<bool>
Testing Span<char>
Testing Span<double>
Testing Span<float>
Testing Span<long>
Testing Span<uint>

Testing Vector64<bool>
Testing Vector64<char>
Testing Vector64<double>
Testing Vector64<float>
Testing Vector64<long>
Testing Vector64<uint>

Testing Vector128<bool>
Gathering state for process 16185 corerun
Crashing thread 00003f39 signal 00000004
Writing minidump with heap to file /home/helixbot/dotnetbuild/dumps/coredump.16185.dmp
Written 67092480 bytes (16380 pages) to core file
Dump successfully written
Expected: 100
Actual: 132
END EXECUTION - FAILED
Test Harness Exitcode is : 1
To run the test:

set CORE_ROOT=/datadisks/disk1/work/BD200A9D/p
/datadisks/disk1/work/BD200A9D/w/B95109F8/e/Interop/PInvoke/Generics/GenericsTest/GenericsTest.sh
Expected: True
Actual:   False


Stack trace
   at Interop_PInvoke._Generics_GenericsTest_GenericsTest_._Generics_GenericsTest_GenericsTest_sh()
```

**Comment by krwq** [comment]
@AndyAyersMS is trying the fix for this: https://github.com/dotnet/runtime/pull/61229
I'm also testing his change in parallel: https://dev.azure.com/dnceng/public/_build/results?buildId=1456069&view=results against full matrix

**Comment by AndyAyersMS** [comment]
Should be fixed by #61229.

**Comment by AaronRobinsonMSFT** [comment]
Fixes with #61229.
